src/components/ContactList.js

```diff
@@ -0,0 +1,102 @@
+import contactData from "../contacts.json";
+import { useState } from 'react';
+import './ContactList.css';
+
+export const ContactList = () => {
+
+    const newList = contactData.slice(0, 5)
+
+    const [contacts, setContacts] = useState(newList);
+
+
+    const addRandomContact = () => {
+        setContacts((prevList) => {
+            const randomContact = contactData[Math.floor(Math.random()*contactData.length)];
+            const existingContact = prevList.find(element => element.id === randomContact.id)
+            if(!existingContact){
+                return prevList.concat(randomContact)
+            } else {
+                addRandomContact();
+            }
+        })
+    }
```

Comment:
Yuu have that bug we discussed in the morning, that app crashes if we click 'add' many times.

One solution would be to use `filter() `instead of` find()`. 
So you want to first create an array where you take list from database and remove all the contacts already displayed. And get random contact from it instead

```
    const addRandomContact = () => {
        setContacts((prevList) => {
            const remainingContacts = contactData.filter((contact)=> {
                return !prevList.includes(contact);
              });

            const randomContact = remainingContacts[Math.floor(Math.random()*remainingContacts.length)];
            return prevList.concat(randomContact)
        })
    }
```